test/functional/feature_fedpeg.py

```diff
@@ -565,28 +565,44 @@ def run_test(self):
-        assert("value" in final_decoded["vout"][2])
-        assert("asset" in final_decoded["vout"][2])
-        blind_pegin_txid = sidechain.sendrawtransaction(pegin_signed["hex"])
-        sidechain.generate(5)
-        assert_equal(sidechain.gettransaction(blind_pegin_txid)["confirmations"], 5)
+        # test with both blindraw and rawblindraw
+        raw_pegin_blinded1 = sidechain.blindrawtransaction(pegin_hex)
+        raw_pegin_blinded2 = sidechain.rawblindrawtransaction(pegin_hex, ["", unspent["amountblinder"]], [10, 15], [unspent["asset"]]*2, ["", unspent["assetblinder"]], "", False)
+        pegin_signed1 = sidechain.signrawtransactionwithwallet(raw_pegin_blinded1)
+        pegin_signed2 = sidechain.signrawtransactionwithwallet(raw_pegin_blinded2)
+        for pegin_signed in [pegin_signed1, pegin_signed2]:
+            final_decoded = sidechain.decoderawtransaction(pegin_signed["hex"])
+            assert(final_decoded["vin"][0]["is_pegin"])
+            assert(not final_decoded["vin"][1]["is_pegin"])
+            assert("assetcommitment" in final_decoded["vout"][0])
+            assert("valuecommitment" in final_decoded["vout"][0])
+            assert("commitmentnonce" in final_decoded["vout"][0])
+            assert("value" not in final_decoded["vout"][0])
+            assert("asset" not in final_decoded["vout"][0])
+            assert(final_decoded["vout"][0]["commitmentnonce_fully_valid"])
+            assert("assetcommitment" in final_decoded["vout"][1])
+            assert("valuecommitment" in final_decoded["vout"][1])
+            assert("commitmentnonce" in final_decoded["vout"][1])
+            assert("value" not in final_decoded["vout"][1])
+            assert("asset" not in final_decoded["vout"][1])
+            assert(final_decoded["vout"][1]["commitmentnonce_fully_valid"])
+            assert("value" in final_decoded["vout"][2])
+            assert("asset" in final_decoded["vout"][2])
+            # check that it is accepted in the mempool
+            accepted = sidechain.testmempoolaccept([pegin_signed["hex"]])[0]
+            if not accepted["allowed"]:
+                print("Blinded pegin not allowed in mempool; reject-reason: %s" % accepted["reject-reason"])
+                raise Exception()
+            print("Blinded transaction looks ok!") # need this print to distinguish failures in for loop
+        # check if they get mined; since we're trying to mine two double spends, disconnect the nodes
+        disconnect_nodes(sidechain, 3)
+        disconnect_nodes(sidechain2, 2)
+        txid1 = sidechain.sendrawtransaction(pegin_signed1["hex"])
+        blocks = sidechain.generate(3)
+        assert_equal(sidechain.getrawtransaction(txid1, True, blocks[0])["confirmations"], 3)
+        txid2 = sidechain2.sendrawtransaction(pegin_signed2["hex"])
+        blocks = sidechain2.generate(3)
+        assert_equal(sidechain2.getrawtransaction(txid2, True, blocks[0])["confirmations"], 3)
```

Comment:
I'd prefer you put the test back in a usable state after this. i.e., re-connect the two nodes at least.

Reply:
hmm yeah I was thinking that it was not needed but someone extending this test should do it. I'll connect them. I was looking for a way to discard the tx from the mempool so I could mine the first one, invalidate block, discard and mine the other one, but I couldn't find how.